features/issues/components/issue-list/issue-list.tsx

```diff
@@ -73,25 +129,46 @@ const PageNumber = styled.span`
 
 export function IssueList() {
   const router = useRouter();
+
+  const [status, setOptionStatus] = useState("");
+  const [level, setOptionLevel] = useState("");
+  const [project, setProjectSearch] = useState("");
+
+  const handleStatusChange = (optionByStatus: any) => {
+    setOptionStatus(optionByStatus.value);
+    router.query.status = optionByStatus.value;
+    router.push(router);
+  };
+  const handleLevelChange = (optionByLevel: any) => {
+    setOptionLevel(optionByLevel.value);
+    router.query.level = optionByLevel.value;
+    router.push(router);
+  };
+  const handleSearchProject = (e: { target: { value: string } }) => {
+    const value = e.target.value.toLocaleLowerCase();
+    setProjectSearch(value);
+    router.query.project = value;
+    router.push(router);
+  };
+
+  const debouncedSearch = useMemo(() => {
+    return debounce(handleSearchProject, 300);
+  }, []);
```

Comment:
Does this work properly? I'd expect the input to update only every 300ms as well. You should debounce the `router.push(router);` call but you can call `setProjectSearch` with every keystroke. Try this (while removing the `setProjectSearch` from the `handleSearchProject` function).

```
  const debouncedSearch = useMemo(() => {
    setProjectSearch(value);
    return debounce(handleSearchProject, 300);
  }, []);
```